### PR TITLE
soroban-rpc: Add metrics for JSON RPC handlers

### PR DESCRIPTION
**Pull request by tamirms**
### What

Add metrics for JSON RPC handlers

### Why

https://github.com/stellar/soroban-tools/issues/496

### Known limitations

N/A

### CODE REVIEW

**Review by tsachiherman — CHANGES_REQUESTED** [review]
Please add a unit test that verify that this work.

File: cmd/soroban-rpc/internal/jsonrpc.go [hunk 1]
```diff
@@ -42,6 +44,44 @@ type HandlerParams struct {
 	Daemon            interfaces.Daemon
 }
 
+type handlerWithMetrics struct {
+	handler       jrpc2.Handler
+	endpoint      string
```
Comment (2opremio) [inline cmd/soroban-rpc/internal/jsonrpc.go:49]: I don't think we need this. Can't we use `Request.Method()`?

File: cmd/soroban-rpc/internal/jsonrpc.go [hunk 2]
```diff
@@ -42,6 +44,44 @@ type HandlerParams struct {
...
+
+func (h *handlerWithMetrics) Handle(ctx context.Context, r *jrpc2.Request) (interface{}, error) {
+	startTime := time.Now()
+	result, err := h.handler.Handle(ctx, r)
+	duration := time.Since(startTime).Seconds()
+	label := prometheus.Labels{"status": "ok"}
+	simulateTransactionResponse, ok := result.(methods.SimulateTransactionResponse)
+	if err != nil || ok && simulateTransactionResponse.Error != "" {
+		label = prometheus.Labels{"status": "error"}
+	}
```
Comment (2opremio) [inline cmd/soroban-rpc/internal/jsonrpc.go:61]: Shouldn't the same be done for the generic JSON-RPC `Response.Error()` field?

https://pkg.go.dev/github.com/creachadair/jrpc2#Response.Error

Comment (2opremio) [reply cmd/soroban-rpc/internal/jsonrpc.go:61]: We may actually want to record the error codes separately

File: cmd/soroban-rpc/internal/test/metrics_test.go [hunk 3]
```diff
@@ -28,5 +22,14 @@ func TestMetrics(t *testing.T) {
 		runtime.Version(),
 		config.Version,
 	)
-	require.Contains(t, responseString, buildMetric)
+	require.Contains(t, metrics, buildMetric)
+}
+
+func getMetrics(test *Test) string {
+	response, err := http.Get(test.adminURL() + "/metrics")
```
Comment (tsachiherman) [inline cmd/soroban-rpc/internal/test/metrics_test.go:29]: this is probably ok here, but just fyi, this is usually being done using "path.Join", which would eliminate double `/`, etc.

**Review by tsachiherman — APPROVED** [review]

File: cmd/soroban-rpc/internal/test/metrics_test.go (hunk 3 above)
Comment (tsachiherman) [reply cmd/soroban-rpc/internal/test/metrics_test.go:29]: sorry - I should have written [`url.JoinPath`](https://pkg.go.dev/net/url@go1.19beta1#JoinPath)

**Pull request merged**